plugin/plugin.go

```diff
@@ -0,0 +1,190 @@
+package plugin
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/PagerDuty/go-pagerduty"
+	"github.com/sirupsen/logrus"
+)
+
+// Severity levels as constants for reuse and readability.
+const (
+	SeverityCritical = "critical"
+	SeverityWarning  = "warning"
+	SeverityInfo     = "info"
+)
+
+// Args provides plugin execution arguments.
+type Args struct {
+	Level             string `envconfig:"PLUGIN_LOG_LEVEL"`
+	RoutingKey        string `envconfig:"PLUGIN_ROUTING_KEY"`
+	IncidentSummary   string `envconfig:"PLUGIN_INCIDENT_SUMMARY"`
+	IncidentSource    string `envconfig:"PLUGIN_INCIDENT_SOURCE"`
+	IncidentSeverity  string `envconfig:"PLUGIN_INCIDENT_SEVERITY"`
```

Comment:
```suggestion
	IncidentSeverity  string `envconfig:"PLUGIN_INCIDENT_SEVERITY"`
```

add a func `validateSeverity` to validate PagerDuty’s allowed severity values (critical, error, warning, info) - check their documentation once for clarity.

and use it in the Exec to validate the user inputted value of severity